core/src/native_fs.rs

```diff
@@ -0,0 +1,151 @@
+use std::fmt::{Debug, Formatter};
+use std::fs;
+use std::io::{stdin, stdout, Write};
+use std::path::Path;
+use std::sync::Arc;
+use wasmer_vfs::host_fs::{File, FileSystem as HostFileSystem};
+use wasmer_vfs::{
+    FileOpener, FileSystem, FsError, Metadata, OpenOptions, OpenOptionsConfig, ReadDir, VirtualFile,
+};
+
+#[derive(Clone)]
+pub struct NativeFs {
+    pub inner: Arc<HostFileSystem>,
+    pub file_opener: NativeFileOpener,
+}
+
+impl Debug for NativeFs {
+    fn fmt(&self, _: &mut Formatter<'_>) -> std::fmt::Result {
+        todo!()
+    }
+}
+
+impl NativeFs {
+    pub fn new() -> Self {
+        NativeFs {
+            inner: Arc::new(HostFileSystem::default()),
+            file_opener: NativeFileOpener::default(),
+        }
+    }
+
+    pub fn config(&mut self, deny_read: bool, deny_write: bool, no_prompts: bool) {
+        self.file_opener.allow_read = !deny_read;
+        self.file_opener.allow_write = !deny_write;
+        self.file_opener.no_prompts = no_prompts;
+    }
+
+    pub fn set_current_pkg(&mut self, name: &str) {
+        self.file_opener.current_pkg = name.to_string();
+    }
+}
+
+impl FileSystem for NativeFs {
+    fn read_dir(&self, path: &Path) -> wasmer_vfs::Result<ReadDir> {
+        self.inner.read_dir(path)
+    }
+
+    fn create_dir(&self, path: &Path) -> wasmer_vfs::Result<()> {
+        self.inner.create_dir(path)
+    }
+
+    fn remove_dir(&self, path: &Path) -> wasmer_vfs::Result<()> {
+        self.inner.remove_dir(path)
+    }
+
+    fn rename(&self, from: &Path, to: &Path) -> wasmer_vfs::Result<()> {
+        self.inner.rename(from, to)
+    }
+
+    fn metadata(&self, path: &Path) -> wasmer_vfs::Result<Metadata> {
+        self.inner.metadata(path)
+    }
+
+    fn remove_file(&self, path: &Path) -> wasmer_vfs::Result<()> {
+        self.inner.remove_file(path)
+    }
+
+    fn new_open_options(&self) -> OpenOptions {
+        OpenOptions::new(Box::new(self.file_opener.clone()))
+    }
+}
+
+#[derive(Clone)]
+pub struct NativeFileOpener {
```

Comment:
I thought that I could leave some input here regarding what we talked about earlier!

To decouple the file system from the system that handles file permissions and prompts the user for access we could create a new trait, maybe named `AccessPolicy` or something like that. This would work very much like the `Resolve` trait that is used by the package manager and be generic over the Context struct: `Context<SomeResolveImpl, SomeAccessPolicyImpl>`

Perhaps the trait could look something like this, although you definitely have more expertise in this area : )
```rs
trait AccessPolicy {
   allowed_to_read(path: &Path /*maybe include which package too*/) -> bool;
   allowed_to_write(path: &Path  /*maybe include which package too*/) -> bool;
}
```

Then we can write multiple implementations for this trait (one for the cli that uses the command-line flags and prompts the user if need be, very much like the code you have already written, and maybe one for the web that allows all operations)